tests/search.py

```diff
@@ -105,3 +206,60 @@ def test_get_units_text_search(units_text_searches):
         # run tests against different qs
         _test_unit_text_search(
             qs, search["text"], search["sfields"], search["exact"])
+
+
+@pytest.mark.django_db
+def test_units_contribution_filter(units_contributor_searches):
+    unit_filter, user = units_contributor_searches
+
+    qs = Unit.objects.all()
+    if not hasattr(UnitContributionFilter, "filter_%s" % unit_filter):
+        with pytest.raises(FilterNotFound):
+            UnitContributionFilter(qs, user=user).filter(unit_filter)
+        return
+    test_qs = [
+        qs,
+        qs.none(),
+        qs.filter(
+            store__translation_project__project=Project.objects.first())]
+    for _qs in test_qs:
+        _test_units_contribution_filter(_qs, user, unit_filter)
+
+
+@pytest.mark.django_db
+def test_units_state_filter(units_state_searches):
+    unit_filter = units_state_searches
+    qs = Unit.objects.all()
+    if not hasattr(UnitStateFilter, "filter_%s" % unit_filter):
+        with pytest.raises(FilterNotFound):
+            UnitStateFilter(qs).filter(unit_filter)
+        return
+    test_qs = [
+        qs,
+        qs.none(),
+        qs.filter(
+            store__translation_project__project=Project.objects.first())]
+    for _qs in test_qs:
+        _test_units_state_filter(_qs, unit_filter)
+
+
+@pytest.mark.django_db
+def test_units_checks_filter(units_checks_searches):
+    check_type, check_data = units_checks_searches
+    qs = Unit.objects.all()
+    test_qs = [
+        qs,
+        qs.none(),
+        qs.filter(
+            store__translation_project__project=Project.objects.first())]
+    for _qs in test_qs:
+        _test_units_checks_filter(_qs, check_type, check_data)
+
+
+@pytest.mark.django_db
+def test_units_checks_filter_bad():
+    qs = Unit.objects.all()
+    with pytest.raises(FilterNotFound):
+        UnitChecksFilter(qs).filter("BAD")
+    # if you dont supply check/category you get empty qs
+    assert not UnitChecksFilter(qs).filter("checks").count()
```

Comment:
neither are being tested - its testing what happens if you dont specify which type of search - ie checks or category

its confusing because the the flag is also called ``checks``